mungegithub/mungers/issue-creator.go

```diff
@@ -0,0 +1,271 @@
+	}
+	creator.allIssues_fetchfreq, _ = time.ParseDuration(fmt.Sprintf("%dm", refreshFreqMins))
+	creator.allIssues_nextfetch = time.Now().Add(creator.allIssues_fetchfreq)
+	return creator
+}
+
+func (creator *IssueCreator) filterValidLabels(candidates []string) []string {
+	if creator.validlabels == nil {
+		//failed to fetch list of valid labels for this repo so trust that labels are valid
+		return candidates
+	}
+	var validCandidates []string
+	for _, l := range candidates {
+		found := false
+		for _, validLabel := range creator.validlabels {
+			if l == *validLabel.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			glog.Errorf("The label '%s' is invalid for the repo '%s/%s'. Dropping label.", l, creator.config.getOrg(), creator.config.getProject())
+		} else {
+			validCandidates = append(validCandidates, l)
+		}
+	}
+	return validCandidates
+}
+
+//Checks if the issue is already on github and tries to sync it if it is not.
+func (creator *IssueCreator) Sync(issue Issue) {
+	creator.refreshIfNeeded()
+
+	//first look for existing issues with this ID
+	id := issue.ID()
+	var closedissues []*github.MungeObject
+	for _, obj := range creator.allIssues {
+		if strings.Contains(*obj.Issue.Body, id) {
+			switch *obj.Issue.State {
+			case "open":
+				//if an open issue is found with the ID then the issue is already synced
+				return
+			case "closed":
+				closedissues = append(closedissues, obj)
+			default:
+				glog.Errorf("Unrecognized issue state for issue #%d. Ignoring this issue.\n", *obj.Issue.Number)
+			}
+		}
+	}
+	//if we get here then no open issues exist for the ID
+	body := issue.Body(closedissues)
+	if body == "" {
+		//Issue indicated that it should not be synced
+		return
+	}
+	if !strings.Contains(body, id) {
+		glog.Fatalf("Programmer error: The following body text does not contain id '%s'.\n%s\n", id, body)
+	}
+
+	title := issue.Title()
+	owners := issue.Owners()
+	labels := issue.Labels()
+	if prio, ok := issue.Priority(); ok {
+		labels = append(labels, "priority/P"+strconv.Itoa(prio))
+	}
+	labels = creator.filterValidLabels(labels)
+
+	//now create the issue
+	if creator.config.getDryRun() {
+		glog.Infof("DryRun--Create Issue:\nTitle:%s\nBody:\n%s\nLabels:%v\nAssigned to:%s\n", title, body, labels, owners)
+		return
+	}
+	obj, err := creator.config.NewIssue(title, body, labels, owners)
+	if err != nil {
+		glog.Errorf("Failed to create a new github issue for issue ID '%s'.\n", id)
+		return
+	}
+	creator.allIssues = append(creator.allIssues, obj)
+}
+
+//refreshes the cache of issues if it is time to
+func (creator *IssueCreator) refreshIfNeeded() {
+	if time.Now().After(creator.allIssues_nextfetch) {
+		if creator.refresh() {
+			creator.allIssues_nextfetch = time.Now().Add(creator.allIssues_fetchfreq)
+		}
+		//if refresh failed we will just use the old data
+	}
+}
+
+func (creator *IssueCreator) refresh() bool {
```

Comment:
We can't call this function every minute, that is prohibitively expensive. issue-cacher.go follows the correct pattern of doing the enumeration once, and then updating using changed issues using the `Munge()` method.